timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -860,64 +915,42 @@ void __global__ k_nonbonded_exclusions(
-
-        RealType beta_dij = real_beta*dij;
-        // RealType ebd = erfc(beta_dij);
-
-        RealType qij = qi*qj;
+        RealType u;
         RealType ebd;
-        RealType es_prefactor = charge_scale*qij*inv_dij*real_es_factor(real_beta, dij, inv_d2ij, ebd);
-
-        // lennard jones
-        RealType inv_d4ij = inv_d2ij*inv_d2ij;
-        RealType inv_d6ij = inv_d4ij*inv_d2ij;
+        RealType es_prefactor;
+        RealType dij;
+        RealType inv_dij;
+        RealType inv_d2ij;
+        compute_electrostatics<RealType, true>(
+            charge_scale,
+            qi,
+            qj,
+            d2ij,
+            beta,
+            dij,
+            inv_dij,
+            inv_d2ij,
+            ebd,
+            es_prefactor,
+            u
+        );
 
         RealType delta_prefactor = es_prefactor;
         RealType real_du_dl = 0;
-
-        RealType u = charge_scale*qij*inv_dij*ebd;
         // lennard jones force
         if(eps_i != 0 && eps_j != 0) {
-            // RealType eps_ij = eps_i * eps_j;
-            // RealType sig_ij = sig_i + sig_j;
-            // RealType sig5 = sig_ij*sig_ij*sig_ij*sig_ij*sig_ij;
-
-            // RealType sig_inv_dij = sig_ij*inv_dij;
-            // RealType sig2_inv_d2ij = sig_inv_dij*sig_inv_dij;
-            // RealType sig6_inv_d6ij = sig2_inv_d2ij*sig2_inv_d2ij*sig2_inv_d2ij;
-            // RealType sig6_inv_d8ij = sig6_inv_d6ij*inv_d2ij;
+            RealType sig_grad;
+            RealType eps_grad;
+            compute_lj<RealType, true>(lj_scale, eps_i, eps_j, sig_i, sig_j, inv_dij, inv_d2ij, u, delta_prefactor, sig_grad, eps_grad);
 
-            RealType eps_ij = eps_i * eps_j;
-            RealType sig_ij = sig_i + sig_j;
-
-            RealType sig_inv_dij = sig_ij*inv_dij;
-            RealType sig2_inv_d2ij = sig_inv_dij*sig_inv_dij;
-            RealType sig4_inv_d4ij = sig2_inv_d2ij*sig2_inv_d2ij;
-            RealType sig6_inv_d6ij = sig4_inv_d4ij*sig2_inv_d2ij;
-            RealType sig6_inv_d8ij = sig6_inv_d6ij*inv_d2ij;
-            RealType sig5_inv_d6ij = sig_ij*sig4_inv_d4ij*inv_d2ij;
-
-            RealType lj_prefactor = lj_scale*eps_ij*sig6_inv_d8ij*(sig6_inv_d6ij*48 - 24);
-            delta_prefactor -= lj_prefactor;
-            u += lj_scale*4*eps_ij*(sig6_inv_d6ij-1)*sig6_inv_d6ij;
-
-            RealType sig_grad = lj_scale*24*eps_ij*sig5_inv_d6ij*(2*sig6_inv_d6ij-1);
-            RealType eps_grad = lj_scale*4*(sig6_inv_d6ij-1)*sig6_inv_d6ij;
-
-            g_sigi += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DSIG>(-sig_grad);
-            g_sigj += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DSIG>(-sig_grad);
-
-            g_epsi += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DEPS>(-eps_grad*eps_j);
-            g_epsj += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DEPS>(-eps_grad*eps_i);
+            g_sigi -= FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DSIG>(sig_grad);
```

Comment:
I noticed that this changed the sign of -= and the term inside FLOAT_TO_FIXED_DU_DP(), was there a reason for this? In general I don't expect the two operations (-= and casting) to bitwise commute... should triple check or revert to old way.

Reply:
Will revert